apps/prairielearn/elements/pl-external-grader-results/pl-external-grader-results.py

```diff
@@ -39,18 +43,18 @@ def ansi_to_html(output):
         return f"[Error converting ANSI to HTML: {e}]\n\n{output}"
 
 
-def prepare(element_html, data):
+def prepare(element_html: str, data: pl.QuestionData) -> None:
     element = lxml.html.fragment_fromstring(element_html)
-    required_attribs = []
-    optional_attribs = []
+    required_attribs: List[str] = []
+    optional_attribs: List[str] = []
     pl.check_attribs(element, required_attribs, optional_attribs)
 
 
-def round_value(val, digits=2):
+def round_value(val: float, digits: int = 2) -> str:
     return format(val, f".{digits}f").rstrip("0").rstrip(".")
 
 
-def render(element_html, data):
+def render(element_html: str, data: pl.QuestionData) -> str:
     if data["panel"] == "submission":
```

Comment:
Instead of putting all of these in a single if statement, you should just return a string immediately if the panel is not `"submission"`. Results in less deeply nested code for the main logic here.

Reply:
[Agreed](https://www.youtube.com/watch?v=CFRhGnuXG-4)!